src/Deferred.ts

```diff
@@ -0,0 +1,122 @@
+// TODO: DOCS!
```

Comment:
Assume that is for another PR.

Reply:
Sort of.. there are existing docs but I wanted to go through and clean them up and add all the `@category` tags where it made sense.

Reply:
Going to assume the clean up is for another PR then :smile: (i.e. won't stop this thread from me approving)

Reply:
I left some comments in there. I think I probably will just copy over the docs we wrote in the `docs/` directory of menu-admin-client